external/Apr1Lib.cmake

```diff
@@ -0,0 +1,160 @@
+                CFLAGS=${aprlib_cflags}
+
+        BUILD_COMMAND
+            make -f Makefile all
+
+        INSTALL_COMMAND
+            make -f Makefile install
+    )
+
+    ExternalProject_Add_Step(Apr1StaticLib unix_post_install
+        COMMENT "Unix/Linux/MacOS Apr1StaticLib install completed"
+        DEPENDEES install
+        ALWAYS 1
+
+        # TODO ZZZ remove these
+        COMMAND echo listing ${LIB_STATIC_APR1_INC_DIR} COMMAND ls ${LIB_STATIC_APR1_INC_DIR}
+        COMMAND echo listing ${LIB_STATIC_APR1_INC_DIR}/apr-1 COMMAND ls ${LIB_STATIC_APR1_INC_DIR}/apr-1
+        COMMAND echo listing ${LIB_STATIC_APR1_LOC} COMMAND ls ${LIB_STATIC_APR1_LOC}
+    )
+
+else()
+    # NOT UNIX - i.e., Windows
+
+    ExternalProject_Add(Apr1StaticLib
+        URL ${aprlib_url}
+
+        UPDATE_COMMAND ""
+        PATCH_COMMAND ""
+
+        CMAKE_GENERATOR ${CMAKE_GENERATOR}
+
+        # TODO Figure out what to do with INSTALL_PDB. We disabled it because
+        # our manual INSTALL_COMMAND was not finding the pdb file and failing.
+        CMAKE_ARGS -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+            -DBUILD_SHARED_LIBS=OFF
+            -DAPR_HAVE_IPV6=OFF
+            -DCMAKE_C_FLAGS=${aprlib_cflags}
+            -DCMAKE_INSTALL_PREFIX=${aprlib_install_prefix}
+            -DINSTALL_PDB=OFF
+
+
+        #LOG_INSTALL 1
+    )
+
+
+    #
+    # Add step to organize generic and architecture-specific apr headers under
+    # include/apr-1 subdirectory
+    # NOTE the unix configure-based installation does that and nupic.core
+    #      depends on this include directory organization.
+    #
+
+    ExternalProject_Add_Step(Apr1StaticLib move_installed_headers_to_apr_1
+        COMMENT "Windows: moving installed apr headers to include/apr-1, as expected by nupic.core"
+
+        DEPENDEES install
+        ALWAYS 0
+        #LOG 1
+
+        # Move the installed ${LIB_STATIC_APR1_INC_DIR}/*.h to
+        # ${LIB_STATIC_APR1_INC_DIR}/apr-1
+        COMMAND
+            ${CMAKE_COMMAND} -DGLOBBING_EXPR=${LIB_STATIC_APR1_INC_DIR}/*.h
+                -DDEST_DIR_PATH=${LIB_STATIC_APR1_INC_DIR}/apr-1
+                -P ${CMAKE_SOURCE_DIR}/external/MoveFilesToNewDir.cmake
+        # Copy <SOURCE_DIR>/include/arch to ${LIB_STATIC_APR1_INC_DIR}/apr-1 as
+        # expected by nupic.core
+        COMMAND
+            ${CMAKE_COMMAND} -E make_directory ${LIB_STATIC_APR1_INC_DIR}/apr-1/arch
+        COMMAND
+            ${CMAKE_COMMAND} -E copy_directory <SOURCE_DIR>/include/arch ${LIB_STATIC_APR1_INC_DIR}/apr-1/arch
+    )
+endif()
+
+
+#
+# Add step to patch apr-1 sources
+#
+
+# Patch file path
+set(aprlib_patch_file "${CMAKE_SOURCE_DIR}/external/common/share/apr/apr.patch")
+
+ExternalProject_Add_Step(Apr1StaticLib patch_sources
+    COMMENT "Patching apr-1 sources"
+
+    DEPENDEES update
+    DEPENDERS configure
+    ALWAYS 0
+    #LOG 1
+
+    COMMAND
+        ${CMAKE_COMMAND} -E echo "Patching <SOURCE_DIR> via ${aprlib_patch_file}"
```

Comment:
Is this necessary? It seems that this duplicates the `COMMENT`, which the documentation says is printed when the step executes.